src/dc_contact.rs

```diff
@@ -659,7 +659,7 @@ pub unsafe fn dc_get_contact_encrinfo(
             });
             ret += &p;
             if self_key.is_none() {
-                dc_ensure_secret_key_exists(context);
+                dc_ensure_secret_key_exists(context).ok();
```

Comment:
What is the purpose of `ok` here if the result is discarded?

Reply:
Because `Result` must be used or rust is unhappy, converting it to an `Option` means it has been used and ignoring the `Option` is fine by rust.  This certainly isn't ideal, but it's fine here because this PR is not refactoring `dc_get_contact_encrinfo()` and the existing code was also ignoring the return value.  When `dc_get_contact_encrinfo()` gets worked on I'd hope this would be cleaned up.

Reply:
How about `#[allow(must_use)]` so we can easily search for `allow` later and eliminate them?

Reply:
Oh, `#[allow(must_use)]` sounds like a decent option too.  But let me suggest something else that has crossed my mind a few times for this as well: we could add a `ResultExt` trait and this could than have a `.yolo()` method, which is an alias for `.unwrap()` but again searchable and with the connotation that it's not fully converted yet.  We could then also have a `.sowat() -> ()` which and does nothing and could be used here.  (better name suggestions welcome, note i already managed to sneak in `CString::yolo()` though, which is also strictly a transitional API)

What do you think?  Would you still prefer the allow approach or do you think the trait approach is interesting too?

As an aside, and not any justification not to make things better, the `.ok()` pattern as well as `.is_ok()` have been used like this before in dcc-rs.  Just like `.unwrap()` covers currently both cases where unwrap is really fine and where it is just transitional.

Reply:
@dignifiedquire any opinion on this?

Reply:
@link2xt what do you think on this?  Which way would you prefer this PR to take?  Let's move this PR along as well :)

Reply:
I would add `must_use` here so we can merge this PR and suggest `sowat` as a separate PR that replaces current use of `ok` as well and maybe adds `yolo`/`sowat` metric to `top_evil_rs.py`.

Reply:
added `#allow(unused_must_use)]` now.